### PR TITLE
MODKBEKBJ-170 API Tests: Providers Endpoint Updates for Tags

### PR DESCRIPTION
**Pull request by stanislav-mureiko**
## Purpose

Updated tests for https://issues.folio.org/browse/MODKBEKBJ-170

## Approach

Updated tests for tags and added new tag schema.

### CODE REVIEW

File: mod-kb-ebsco/mod-kb-ebsco-java.postman_collection.json [hunk 1]
```diff
@@ -149,7 +153,7 @@
 						"header": [
 							{
 								"key": "X-Okapi-Module-Id",
-								"value": "{{kb-ebsco-java-module-id}}",
+								"value": "mod-kb-ebsco-java-2.0.4-SNAPSHOT",
```
Comment (natali-zaitseva) [inline mod-kb-ebsco/mod-kb-ebsco-java.postman_collection.json]: it would be good not to hardcode version of the module but use the environment variable `{{kb-ebsco-java-module-id}}` to retrieve the version installed

File: mod-kb-ebsco/mod-kb-ebsco-java.postman_collection.json [hunk 2]
```diff
@@ -7377,7 +7724,7 @@
 											"raw": ""
 										},
 										"url": {
-											"raw": "{{protocol}}://{{url}}:{{okapiport}}/eholdings/packages/{{packageId}}?include=",
+											"raw": "{{protocol}}://{{url}}:{{okapiport}}/eholdings/packages/{{packageId}}?include=prov",
```
Comment (natali-zaitseva) [inline mod-kb-ebsco/mod-kb-ebsco-java.postman_collection.json]: let's remove the `prov` parameter for test name consistency

File: mod-kb-ebsco/mod-kb-ebsco-java.postman_collection.json [hunk 3]
```diff
@@ -118,12 +118,17 @@
 									"   }else{",
 									"       reject(err, echoGetRequest);",
 									"   }",
-									"      }, 5000);",
+									"      }, 7000);",
```
Comment (carolegodfrey) [inline mod-kb-ebsco/mod-kb-ebsco-java.postman_collection.json]: Just curious as to why this timeout needed to be increased? Wonder if environment variable may be a good place for holding this if this needs to be tweaked frequently. Although this is the only place it is referenced

File: mod-kb-ebsco/mod-kb-ebsco-java.postman_collection.json [hunk 4]
```diff
@@ -1243,7 +1247,139 @@
...
-							"    tv4.addSchema(\"schema_jsonapi.json\", JSON.parse(pm.variables.get(\"schema_jsonapi\")));"
+							"    tv4.addSchema(\"schema_jsonapi.json\", JSON.parse(pm.variables.get(\"schema_jsonapi\")));",
+							"    tv4.addSchema(\"schema_tags.schema\", JSON.parse(pm.variables.get(\"schema_tags\")));"
+						]
+					}
+				}
+			]
+		},
+		{
+			"name": "setup for providers adding tags",
```
Comment (carolegodfrey) [inline mod-kb-ebsco/mod-kb-ebsco-java.postman_collection.json]: Just wondering why this `setup for provider with adding tags`  is needed? Its looks like we are testing tags as part of PUT request (both adding and removing tags as part of these tests) - the tags added in this setup step are not checked for anywhere. And because there is a setup step, I was looking for a matching teardown step.

File: mod-kb-ebsco/mod-kb-ebsco-java.postman_collection.json (hunk 3 above)
Comment (stanislav-mureiko) [reply mod-kb-ebsco/mod-kb-ebsco-java.postman_collection.json]: Tests pass twice. One time via Postman, second time via Newman.This timeout needed for uploading all schemes for tests. With Postman it is okay. Newman does not upload all schemes for that time. As solution I have increased timeout for uploading schemes.

File: mod-kb-ebsco/mod-kb-ebsco-java.postman_collection.json (hunk 4 above)
Comment (stanislav-mureiko) [reply mod-kb-ebsco/mod-kb-ebsco-java.postman_collection.json]: We are not able to create and delete new provider. And we have not had providers with existing tags. As setup step I use updating existing provider by adding new tags.
We do not have teardown step because tags is new functionality and it should be added to providers as a new attribute of this object.

File: mod-kb-ebsco/mod-kb-ebsco-java.postman_collection.json (hunk 3 above)
Comment (carolegodfrey) [reply mod-kb-ebsco/mod-kb-ebsco-java.postman_collection.json]: Ok thanks @stanislav-mureiko for the info

File: mod-kb-ebsco/mod-kb-ebsco-java.postman_collection.json (hunk 4 above)
Comment (carolegodfrey) [reply mod-kb-ebsco/mod-kb-ebsco-java.postman_collection.json]: Thanks Stan,  I understand all this -- my question/comment was that I  don't see anywhere in the tests where you are using and testing for foo-tag that that was added in this setup step -- that is why I am questioning  it. For example  - with `PUT update provider with tags` you set tags and you check for them and then there is another test that tests an empty tag list . Which test is using the setup?

Comment (stanislav-mureiko) [reply mod-kb-ebsco/mod-kb-ebsco-java.postman_collection.json]: Thanks Carole,it makes sense to test adding tags in setup step.

Comment (carolegodfrey) [reply mod-kb-ebsco/mod-kb-ebsco-java.postman_collection.json]: Thanks Stas -- I think this is fine -- I was questioning the setup step because it didn't look like the setup for providers tags was used elsewhere by other tests
